kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/utils/TestUtils.scala

```diff
@@ -797,27 +798,37 @@ trait TestUtils extends Assertions with SQLHelper {
    * Verify checksum data matches the expected values in the snapshot.
    * @param snapshot Snapshot to verify the checksum against
    */
-  protected def verifyChecksumForSnapshot(snapshot: Snapshot): Unit = {
-    val crcInfo = ChecksumReader.getCRCInfo(
+  protected def verifyChecksumForSnapshot(
+      snapshot: Snapshot,
+      expectEmptyTable: Boolean = false): Unit = {
+    val crcInfoOpt = ChecksumReader.getCRCInfo(
       defaultEngine,
       new KernelPath(snapshot.asInstanceOf[SnapshotImpl].getLogPath.toString),
       snapshot.getVersion,
       snapshot.getVersion)
     assert(
-      crcInfo.isPresent,
+      crcInfoOpt.isPresent,
       s"CRC information should be present for version ${snapshot.getVersion}")
-    // TODO: check metadata, protocol and file size.
-    assert(
-      crcInfo.get().getNumFiles === collectScanFileRows(snapshot.getScanBuilder.build()).size,
-      "Number of files in checksum should match snapshot")
-    // CRC does not store tombstones.
-    assert(
-      crcInfo.get().getDomainMetadata === Optional.of(
-        snapshot.asInstanceOf[SnapshotImpl].getDomainMetadataMap.values().asScala
-          .filterNot(_.isRemoved)
-          .toSet
-          .asJava),
-      "Domain metadata in checksum should match snapshot")
+    crcInfoOpt.toScala.foreach { crcInfo =>
+      // TODO: check metadata, protocol and file size.
+      assert(
+        crcInfo.getNumFiles === collectScanFileRows(snapshot.getScanBuilder.build()).size,
+        "Number of files in checksum should match snapshot")
+      if (expectEmptyTable) {
```

Comment:
this is a super easy additional check to add for the tableSizeInBytes and fileSizeHistogram; in the future we should probably revisit this but this was a super low-effort improvement and it make sure these are reset and 0/empty for replace table (without ctas)